src/VisualStudio/Core/Def/Implementation/CodeLens/UnitTestingReferencesService.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeLens;
+using Microsoft.VisualStudio.Language.CodeLens;
+using Microsoft.VisualStudio.Language.CodeLens.Remoting;
+using Microsoft.VisualStudio.LanguageServices.CodeLens;
+
+namespace Microsoft.CodeAnalysis.UnitTesting.ExternalAccess
+{
+    internal class UnitTestingReferencesService
```

Comment:
@heejaechang @sharwell FYI I synced up with @mavasani about external access and have tried to follow the new pattern that was discussed today for this. We will obviously need to revisit this once the analyzer is ready - but please let me know if you have feedback.

Reply:
The analyzer approach to external access layers only applies to shipping assemblies. Services exposed for testing purposes can use the external access assembly which is already in source control:
https://github.com/dotnet/roslyn/tree/master/src/Tools/ExternalAccess/UnitTesting

Reply:
Sorry but this will result in additional assembly load. Although the external access layer for UnitTesting exists, nothing is consuming it at the moment.

Reply:
Ah @sharwell the service is not exposed for testing purposes. We want to consume this service in the shipping product (i.e. the implementation of the new tested by code lens indicator which is being added in unit testing repo).